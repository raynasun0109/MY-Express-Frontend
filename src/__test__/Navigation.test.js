// import renderer from 'react-test-renderer';
// import Navigation from '../components/Navigation/Navigation.js';
// import { renderWithProviders } from "../utils/utils-for-tests";

// const initState={
//     totalNumber:0,
//     shoppingCart:[],
// }
// it('Render Navigation Component', () => {
//     const component = renderer.create(
//         <renderWithProviders>
//         <Navigation data={initState}/>
//         </renderWithProviders>
//     );
//     let tree = component.toJSON();
//     expect(tree).toMatchSnapshot();

// })

// // test("Render Navigation Home and About Us and Login and register", async () => {
// //     render(<Navigation data={initState}/>)
// //     const NavigationComponent = screen.getByTestId("navigation");
// //     expect(NavigationComponent).toHaveTextContent("Home");
// //     expect(NavigationComponent).toHaveTextContent("About Us");
// //     expect(NavigationComponent).toHaveTextContent("Login");
// //     expect(NavigationComponent).toHaveTextContent("Register");
// // });

