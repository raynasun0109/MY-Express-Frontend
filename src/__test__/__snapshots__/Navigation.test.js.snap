// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Navigation component renders correctly 1`] = `
<DocumentFragment>
  <div>
    <header
      class="MuiPaper-root MuiPaper-elevation MuiPaper-elevation4 MuiAppBar-root MuiAppBar-colorPrimary MuiAppBar-positionStatic navigation_container css-hip9hq-MuiPaper-root-MuiAppBar-root"
    >
      <div
        class="MuiContainer-root MuiContainer-maxWidthXl css-19r6kue-MuiContainer-root"
      >
        <div
          class="MuiToolbar-root MuiToolbar-regular css-r6ewbb-MuiToolbar-root"
        >
          <div
            class="MuiBox-root css-1lvtzne"
          >
            <div>
              <button
                class="MuiButtonBase-root MuiButton-root MuiButton-text MuiButton-textPrimary MuiButton-sizeMedium MuiButton-textSizeMedium MuiButton-root MuiButton-text MuiButton-textPrimary MuiButton-sizeMedium MuiButton-textSizeMedium css-1e6y48t-MuiButtonBase-root-MuiButton-root"
                data-testid="drawer-button"
                tabindex="0"
                type="button"
              >
                <svg
                  aria-hidden="true"
                  class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium navigation_container_menu_icon css-i4bv87-MuiSvgIcon-root"
                  data-testid="MenuRoundedIcon"
                  focusable="false"
                  viewBox="0 0 24 24"
                >
                  <path
                    d="M4 18h16c.55 0 1-.45 1-1s-.45-1-1-1H4c-.55 0-1 .45-1 1s.45 1 1 1zm0-5h16c.55 0 1-.45 1-1s-.45-1-1-1H4c-.55 0-1 .45-1 1s.45 1 1 1zM3 7c0 .55.45 1 1 1h16c.55 0 1-.45 1-1s-.45-1-1-1H4c-.55 0-1 .45-1 1z"
                  />
                </svg>
                <span
                  class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
                />
              </button>
            </div>
          </div>
          <div
            class="MuiBox-root css-1t6c9ts"
          >
            <button
              class="MuiButtonBase-root MuiButton-root MuiButton-text MuiButton-textPrimary MuiButton-sizeMedium MuiButton-textSizeMedium MuiButton-root MuiButton-text MuiButton-textPrimary MuiButton-sizeMedium MuiButton-textSizeMedium desktop_nav_content css-1q39md6-MuiButtonBase-root-MuiButton-root"
              data-testid="Home-button"
              tabindex="0"
              type="button"
            >
              Home
              <span
                class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
              />
            </button>
            <button
              class="MuiButtonBase-root MuiButton-root MuiButton-text MuiButton-textPrimary MuiButton-sizeMedium MuiButton-textSizeMedium MuiButton-root MuiButton-text MuiButton-textPrimary MuiButton-sizeMedium MuiButton-textSizeMedium desktop_nav_content css-1q39md6-MuiButtonBase-root-MuiButton-root"
              data-testid="About Us-button"
              tabindex="0"
              type="button"
            >
              About Us
              <span
                class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
              />
            </button>
          </div>
          <div
            class="cart_container MuiBox-root css-0"
          >
            <a
              href="/dashboard/user/undefined/carts"
            >
              <span
                class="MuiBadge-root cart_container_badge css-1c32n2y-MuiBadge-root"
              >
                <svg
                  aria-hidden="true"
                  class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium cart_container_cart css-i4bv87-MuiSvgIcon-root"
                  data-testid="ShoppingCartIcon"
                  focusable="false"
                  viewBox="0 0 24 24"
                >
                  <path
                    d="M7 18c-1.1 0-1.99.9-1.99 2S5.9 22 7 22s2-.9 2-2-.9-2-2-2zM1 2v2h2l3.6 7.59-1.35 2.45c-.16.28-.25.61-.25.96 0 1.1.9 2 2 2h12v-2H7.42c-.14 0-.25-.11-.25-.25l.03-.12.9-1.63h7.45c.75 0 1.41-.41 1.75-1.03l3.58-6.49c.08-.14.12-.31.12-.48 0-.55-.45-1-1-1H5.21l-.94-2H1zm16 16c-1.1 0-1.99.9-1.99 2s.89 2 1.99 2 2-.9 2-2-.9-2-2-2z"
                  />
                </svg>
                <span
                  class="MuiBadge-badge MuiBadge-standard MuiBadge-invisible MuiBadge-anchorOriginTopRight MuiBadge-anchorOriginTopRightRectangular MuiBadge-overlapRectangular css-1s52at1-MuiBadge-badge"
                />
              </span>
            </a>
          </div>
          <div
            class="MuiBox-root css-2uchni"
          >
            <div
              class="login_register_container"
            >
              <a
                href="/login"
              >
                Login
              </a>
              /
              <a
                href="/register"
              >
                Register
              </a>
            </div>
          </div>
        </div>
      </div>
    </header>
  </div>
</DocumentFragment>
`;
